autosklearn/estimators.py

```diff
@@ -31,7 +44,8 @@ def __init__(
         time_left_for_this_task=3600,
         per_run_time_limit=None,
         initial_configurations_via_metalearning=25,
-        ensemble_size: int = 50,
+        ensemble_class: Type[AbstractEnsemble] | None = EnsembleSelection,
+        ensemble_kwargs: Dict[str, Any] | None = None,
```

Comment:
This is the main API breaking change, I imagine most power users will have `ensemble_size` specified and maybe we should deprecate it, maintaining old default behavior by inserting it into `ensemble_kwargs`? We can issue a warning:

```python
# In function defnition
    ensemble_size: int | None = None
    

# User specified `ensemble_size` explicitly, warn them about deprecation
if ensemble_size is not None: 
    warnings.warn(f"`ensemble_size` has been deprecated, please use `ensemble_kwargs = { 'ensemble_size': {ensemble_size} }. Inserting `ensemble_size` into `ensemble_kwargs` for now.")
    # Keep consistent behaviour
    if ensemble_kwargs is None:
        ensemble_kwargs = {"ensemble_size": ensemble_size}
    else:
        ensemble_kwargs["ensemble_size"] = ensemble_size
else:
    # Old default behaviour, no need to warn here as they havn't set `ensemble_size`
    if ensemble_kwargs is None:
        ensemble_kwargs = {"ensemble_size": 50}
```

Porbably not the cleanest solution but hopefully it gets the point across.

Reply:
As mentioned above I'll deprecate this. I'll deprecate this here so that automl.py will only accept ensemle_kwargs

Reply:
Updated this.